src/modules/_[spaceId]/HomePage/HomePageNavbar.tsx

```diff
@@ -0,0 +1,140 @@
+import Button from '@/components/Button'
+import Input from '@/components/inputs/Input'
+import useWrapInRef from '@/hooks/useWrapInRef'
+import { cx } from '@/utils/class-names'
+import { useEffect, useRef, useState } from 'react'
+import { BsXCircleFill } from 'react-icons/bs'
+import { HiMagnifyingGlass } from 'react-icons/hi2'
+
+export type HomePageNavbarProps = {
+  logo: JSX.Element
+  auth: JSX.Element
+  colorModeToggler: JSX.Element
+  search: string
+  setSearch: (search: string) => void
+  setFocusedElementIndex: React.Dispatch<React.SetStateAction<number>>
+  onUpClick: () => void
+  onDownClick: () => void
+}
+
+export default function HomePageNavbar({
+  auth,
+  colorModeToggler,
+  logo,
+  search,
+  setSearch,
+  onDownClick,
+  onUpClick,
+}: HomePageNavbarProps) {
+  const [openSearch, setOpenSearch] = useState(false)
+  const searchRef = useRef<HTMLInputElement | null>(null)
+
+  const clearOrCloseSearch = useWrapInRef(() => {
+    if (search) {
+      setSearch('')
+      searchRef.current?.focus()
+    } else {
+      setOpenSearch(false)
+    }
+  })
+
+  useEffect(() => {
+    const keyListener = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        clearOrCloseSearch.current()
+      }
+    }
+    window.addEventListener('keydown', keyListener)
+    return () => window.removeEventListener('keydown', keyListener)
+  }, [clearOrCloseSearch])
```

Comment:
We are using a lot of hot keys, maybe better start using a library special for hotkeys?

Like this:
https://www.npmjs.com/package/react-hotkeys-hook